### DISCUSSION TITLE
Is it ok to create custom hooks that use atoms inside and return values?

### DISCUSSION THREAD

**Discussion by azabraao** [pmndrs/jotai#1618] [Q&A] [answered]
Instead of using the `useAtom` in my react components, I'm choosing to create custom hooks for trying to bring more semantics to the code, using custom hooks, keeping the atom-related logic all inside it. This way I'm expecting that `useUser()` becomes a little more obvious than `useAtom(user)` when others read the code.

So I am wondering what would be the downside of this approach:

```javascript

const userAtom = atom({});

export default function useUser() {
  const [user, setUser] = useAtom(userAtom);

  useEffect(() => {
    // as a user fetch
    const hasUser = Object.keys(user).length;
    if (!hasUser) setUser({ name: "fake" });
  }, []);

  return user;
}

```

Is it recommended? Is there any better practice?

**Comment by dai-shi** [comment, marked as answer]
I assume the question is "Is it ok to create custom hooks that use atoms inside and return values?"

Yes, what you are doing is totally fine.

**Reply by azabraao** [reply]
Oh, yeah, this is what I meant. I'm updating the question title. Thank you.
